src/tests/stats/test_distribution_uniform.fypp

```diff
@@ -0,0 +1,298 @@
+        #:elif t1[0] == "r"
+        loc = 0.0_${k1}$; scale = 2.0_${k1}$
+        #:else
+        loc = (-0.5_${k1}$, 0.5_${k1}$); scale = (1.0_${k1}$, 1.0_${k1}$)
+        #:endif
+        x1 = uni_rvs(loc, scale)
+        x2 = reshape(uni_rvs(loc, scale, 12), [3,4])
+        res(:,1) = uni_pdf(x1, loc, scale)
+        res(:, 2:5) = uni_pdf(x2, loc, scale)
+        call check(all(abs(res - reshape(ans,[3,5])) < sptol),                 &
+            msg = "uniform_distribution_pdf_${t1[0]}$${k1}$ failed", warn=warn)
+    end subroutine test_uni_pdf_${t1[0]}$${k1}$
+
+    #:endfor
+
+
+
+    #:for k1, t1 in ALL_KINDS_TYPES
+    subroutine test_uni_cdf_${t1[0]}$${k1}$
+        ${t1}$ :: x1, x2(3,4), loc, scale
+        real :: res(3,5)
+        integer :: seed, get
+        #:if k1 == "int8"
+        real, parameter :: ans(15) = [0.435643554, 0.435643554, 0.435643554,   &
+                                      0.702970326, 0.653465331, 0.485148519,   &
+                                      0.386138618, 0.386138618, 0.336633652,   &
+                                      0.277227730, 0.237623766, 0.524752498,   &
+                                      0.732673287, 0.534653485, 0.415841579]
+        #:elif k1 == "int16"
+        real, parameter :: ans(15) = [0.178217828, 0.178217828, 0.178217828,   &
+                                      0.465346545, 0.673267305, 0.247524753,   &
+                                      0.158415839, 0.792079210, 0.742574275,   &
+                                      0.574257433, 0.881188095, 0.663366318,   &
+                                      0.524752498, 0.623762369, 0.178217828]
+        #:elif k1 == "int32"
+        real, parameter :: ans(15) = [0.732673287, 0.732673287, 0.732673287,   &
+                                      0.722772300, 0.792079210, 5.94059415E-02,&
+                                      0.841584146, 0.405940592, 0.960396051,   &
+                                      0.534653485, 0.782178223, 0.861386120,   &
+                                      0.564356446, 0.613861382, 0.306930691]
+        #:elif k1 == "int64"
+        real, parameter :: ans(15) = [0.455445558, 0.455445558, 0.455445558,   &
+                                      0.277227730, 0.455445558, 0.930693090,   &
+                                      0.851485133, 0.623762369, 5.94059415E-02,&
+                                      0.693069279, 0.544554472, 0.207920790,   &
+                                      0.306930691, 0.356435657, 0.128712878]
+        #:elif t1[0] == "r"
+        real, parameter :: ans(15) = [0.170192942, 0.170192942, 0.170192942,   &
+                                      0.276106149, 0.754930079, 0.406620681,   &
+                                      0.187742814, 0.651605546, 0.934733927,   &
+                                      0.151271492, 0.987674534, 0.130533904,   &
+                                      0.106271908, 9.27578658E-02, 0.203399420]
+        #:else
```

Comment:
I'm not immediately sure which type this case treats, again suggest to make the fypp slightly more explicit.